plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/cache/DeltaLakeAlluxioFileSystemCache.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.cache;
+
+import com.google.inject.Inject;
+import io.trino.filesystem.TrinoInput;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+import io.trino.filesystem.alluxio.AlluxioFileSystemCache;
+import io.trino.filesystem.cache.TrinoFileSystemCache;
+
+import java.io.IOException;
+
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeAlluxioFileSystemCache
+        implements TrinoFileSystemCache
+{
+    private final AlluxioFileSystemCache cache;
+
+    @Inject
+    public DeltaLakeAlluxioFileSystemCache(AlluxioFileSystemCache cache)
+    {
+        this.cache = requireNonNull(cache, "cache is null");
+    }
+
+    @Override
+    public TrinoInput cacheInput(TrinoInputFile delegate)
+            throws IOException
+    {
+        // TODO: Enable caching of parquet checkpoint files once the CheckpointEntryIterator is always closed
```

Comment:
maybe instead of a separate implementation, we could bind a connector-specific filter predicate, that will bypass caching for locations matching a pattern or any other condition?

Something like `boolean isCacheable(TrinoInputFile)`

Reply:
We can have more sophisticated logic than, not only bypass cache based on a name, but size or other conditions

Reply:
Went with `Optional<String> getCacheKey(TrinoInputFile)`